drupal/Drupal.py

```diff
@@ -176,17 +176,18 @@ def prepare_database(repo, branch, build, buildtype, alias, site, syncbranch, or
   # If sync_branch_host and current_env match, we don't need to connect to another
   # server to get the dump
   if sync_branch_host == current_env:
-    # Check a site exists on this server
-    if run('drush sa | grep \'^@\?%s_%s$\' > /dev/null' % (alias, syncbranch)).failed:
-      raise SystemError("######## Cannot find a site with the alias %s_%s. Aborting." % (alias, syncbranch))
+    # Check that the syncbranch site exists on this server
+    syncbranch_site = common.Utils.get_previous_build(repo, syncbranch, None)
+    if syncbranch_site is None:
+      raise SystemError("######## Cannot find a site to sync the database from. Aborting.")
 
     # If freshinstall is True, this occurs during the initial build, so we create a new database
     # dump in the db/ directory which will be imported
     if freshinstall:
       print "===> Database to get a fresh dump from is on the same server. Getting database dump now..."
       # Time to dump the database and save it to db/
       dump_file = "%s_%s.sql.bz2" % (alias, syncbranch)
-      run('drush @%s_%s sql-dump | bzip2 -f > /var/www/%s_%s_%s/db/%s' % (alias, syncbranch, repo, branch, build, dump_file))
+      run('cd /var/www/live.%s.%s/www/sites/%s && drush -y sql-dump | bzip2 -f > /var/www/%s_%s_%s/db/%s' % (repo, syncbranch, site, repo, branch, build, dump_file))
```

Comment:
This can use the drush wrapper function I made, no?

Reply:
Discussed in IRC - cannot use wrapper because of piping the output to bzip2.